### PR TITLE
[CPU] CumSum: dynamic shapes support

### PR DESCRIPTION
**Pull request by v-Golubev**
### Details:
 - *CumSum node: dynamic shapes support*
 - *CumSum: added CPU tests with dynamic shapes*

### Tickets:
 - *69882*

### CODE REVIEW

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp [hunk 1]
```diff
@@ -45,7 +41,7 @@ MKLDNNCumSumNode::MKLDNNCumSumNode(const std::shared_ptr<ngraph::Node>& op, cons
     if ((getOriginalInputsNumber() != numOfInputs && getOriginalInputsNumber() != (numOfInputs - 1)) || getOriginalOutputsNumber() != 1)
         IE_THROW() << errorPrefix << " has incorrect number of input/output edges!";
 
-    const auto &dataShape = op->get_input_shape(CUM_SUM_DATA);
+    const auto &dataShape = op->get_input_partial_shape(CUM_SUM_DATA);
```
Comment (mandrono) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp:44]: use `getInputShapeAtPort/getOutputShapeAtPort`

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp [hunk 2]
```diff
@@ -162,6 +157,7 @@ template <bool reverse, bool exclusive, typename dataType>
 void MKLDNNCumSumNode::cumSum(const dataType *input, dataType *output, const VectorDims &strides) {
     SizeVector iterationRange(numOfDims - 1);
     size_t j = 0;
+    SizeVector shape = getParentEdgesAtPort(CUM_SUM_DATA)[0]->getMemory().getStaticDims();
```
Comment (mandrono) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp:160]:
```suggestion
    const auto &shape = getParentEdgesAtPort(CUM_SUM_DATA)[0]->getMemory().getStaticDims();
```

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+using namespace ngraph;
+using namespace InferenceEngine;
+using namespace CPUTestUtils;
+
+namespace CPULayerTestsDefinitions {
+
+using cumSumParams = std::tuple<
+    ngraph::element::Type, // data precision
+    std::pair<std::vector<ngraph::PartialShape>, std::vector<std::vector<ngraph::Shape>>>, // input shape
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp:17]: make alias for this smt like `CumSumShape`

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+const std::vector<bool> exclusive = { true, false };
+const std::vector<bool> reverse = { true, false };
+
+using shapesSuite = std::vector<std::pair<std::vector<ngraph::PartialShape>, std::vector<std::vector<ngraph::Shape>>>>;
+
+shapesSuite inShapes = {
+    {
+        // dynamic
+        {
+            {-1}
```
Comment (mandrono) [inline inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp:98]: add test cases with upper bounds

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp (hunk 1 above)
Comment (v-Golubev) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp]: Done, thanks

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp (hunk 2 above)
Comment (v-Golubev) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp]: Corrected

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp (hunk 3 above)
Comment (v-Golubev) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp]: Thanks, done

File: inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp (hunk 4 above)
Comment (v-Golubev) [reply inference-engine/tests/functional/plugin/cpu/single_layer_tests/cum_sum.cpp:97]: Test cases with lower/upper bounds was added, thank you.

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp [hunk 5]
```diff
@@ -45,11 +41,11 @@ MKLDNNCumSumNode::MKLDNNCumSumNode(const std::shared_ptr<ngraph::Node>& op, cons
...
 
-    const auto &dataShape = op->get_input_shape(CUM_SUM_DATA);
-    if (dataShape.size() < 1) {
-        IE_THROW() << errorPrefix << " doesn't support 'data' input tensor with rank: " << dataShape.size();
+    const auto &dataShape = getInputShapeAtPort(CUM_SUM_DATA);
+    if (dataShape.getRank() < 1) {
+        IE_THROW() << errorPrefix << " doesn't support 'data' input tensor with rank: " << dataShape.getRank();
     }
-    numOfDims = dataShape.size();
+    numOfDims = dataShape.getRank();
```
Comment (mandrono) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp:48]: minor
maybe move above and use instead `dataShape.getRank()` in comparasion

**Review by mandrono — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp (hunk 5 above)
Comment (v-Golubev) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_cum_sum_node.cpp:45]: Fixed, thank you

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**